src/file_parsing.jl

```diff
@@ -0,0 +1,327 @@
+end
+
+
+"""
+    log_parameter_information(param_set::ParamDict{FT}, filepath; warn_else_error = "warn") where {FT}
+
+Writes the parameter log file at `filepath`; checks that override parameters are all used.
+"""
+function log_parameter_information(param_set::ParamDict{FT}, filepath; warn_else_error = "warn") where {FT}
+    #[1.] write the parameters to log file
+    write_log_file(param_set,filepath)
+    #[2.] send warnings or errors if parameters were not used
+    check_override_parameter_usage(param_set,warn_else_error)
+end
+
+
+"""
+    merge_override_default_values(override_param_struct::ParamDict{FT},default_param_struct::ParamDict{FT}) where {FT}
+
+Combines the `default_param_struct` with the `override_param_struct`, precedence is given to override information.
+"""
+function merge_override_default_values(override_param_struct::ParamDict{FT},default_param_struct::ParamDict{FT}) where {FT}
+    data = default_param_struct.data
+    dict_type = default_param_struct.dict_type
+    override_dict = override_param_struct.override_dict
+    for (key, val) in override_param_struct.data
+        if ~(key in keys(data))
+            data[key] = val
+        else
+            for (kkey,vval) in val # as val is a Dict too
+                data[key][kkey] = vval
+            end
+        end
+    end
+    return ParamDict{FT}(data, dict_type, override_dict)
+end
+
+"""
+    create_parameter_struct(path_to_override, path_to_default; dict_type="alias", value_type=Float64)
+
+Creates a `ParamDict{value_type}` struct, by reading and merging upto two TOML files with override information taking precedence over default information.
+"""
+function create_parameter_struct(path_to_override, path_to_default; dict_type="alias", value_type=Float64)
+    #if there isn't  an override file take defaults
+    if isnothing(path_to_override)
+        return ParamDict{value_type}(parse_toml_file(path_to_default), dict_type, nothing)
+    else
+        try 
+            override_param_struct = ParamDict{value_type}(parse_toml_file(path_to_override), dict_type, parse_toml_file(path_to_override))
+            default_param_struct = ParamDict{value_type}(parse_toml_file(path_to_default), dict_type, nothing)
+        
+            #overrides the defaults where they clash
+            return merge_override_default_values(override_param_struct, default_param_struct)
+        catch
+            @warn("Error in building from parameter file: "*"\n " * path_to_override * " \n instead, created using defaults from CLIMAParameters...")
+            return ParamDict{value_type}(parse_toml_file(path_to_default), dict_type, nothing)
+        end
+    end
+        
+end
+
+
+"""
+    create_parameter_struct(path_to_override; dict_type="alias", value_type=Float64)
+
+a single filepath is assumed to be the override file, defaults are obtained from the CLIMAParameters defaults list.
+"""
+function create_parameter_struct(path_to_override; dict_type="alias", value_type=Float64)
```

Comment:
```suggestion
function create_parameter_struct(path_to_override; dict_type="alias", ::Type{float_type}) where {float_type <: AbstractFloat}
```

Reply:
Done, though moved it into positional argument